node_test/reverseproxy_test.go

```diff
@@ -90,28 +116,60 @@ func TestReversePaymentProxy(t *testing.T) {
 	checkResponse(t, resp, smallResponse, http.StatusOK)
 
 	// It should properly handle a request to a non existent endpoint
 	voucher = createVoucher(t, aliceClient, paymentChannel, 5)
-	resp = performGetRequest(t, fmt.Sprintf("http://%s/badpath?channelId=%s&amount=%d&signature=%s", proxyAddress, voucher.ChannelId, voucher.Amount.Uint64(), voucher.Signature.ToHexString()))
+	resp = performGetRequest(t, "", fmt.Sprintf("http://%s/badpath?channelId=%s&amount=%d&signature=%s", proxyAddress, voucher.ChannelId, voucher.Amount.Uint64(), voucher.Signature.ToHexString()))
 	checkResponse(t, resp, "", http.StatusNotFound)
 
 	// It should return a larger response if the voucher is large enough
-	voucher = createVoucher(t, aliceClient, paymentChannel, 20)
-	resp = performGetRequest(t, fmt.Sprintf("http://%s/largeResource?channelId=%s&amount=%d&signature=%s", proxyAddress, voucher.ChannelId, voucher.Amount.Int64(), voucher.Signature.ToHexString()))
-	checkResponse(t, resp, largeResponse, http.StatusOK)
+	voucher = createVoucher(t, aliceClient, paymentChannel, uint64(len(testFileContent)))
+	resp = performGetRequest(t, "", fmt.Sprintf("http://%s/file?channelId=%s&amount=%d&signature=%s", proxyAddress, voucher.ChannelId, voucher.Amount.Int64(), voucher.Signature.ToHexString()))
+	checkResponse(t, resp, testFileContent, http.StatusOK)
 
 	// It should return a payment required response for a large response if the voucher is  not large enough
 	voucher = createVoucher(t, aliceClient, paymentChannel, 5)
-	resp = performGetRequest(t, fmt.Sprintf("http://%s/largeResource?channelId=%s&amount=%d&signature=%s", proxyAddress, voucher.ChannelId, voucher.Amount.Int64(), voucher.Signature.ToHexString()))
-	checkResponse(t, resp, expectedPaymentErrorMessage(20, 5), http.StatusPaymentRequired)
+	resp = performGetRequest(t, "", fmt.Sprintf("http://%s/file?channelId=%s&amount=%d&signature=%s", proxyAddress, voucher.ChannelId, voucher.Amount.Int64(), voucher.Signature.ToHexString()))
+	checkResponse(t, resp, expectedPaymentErrorMessage(len(testFileContent), 5), http.StatusPaymentRequired)
+
+	// It should handle a simple range request
+	voucher = createVoucher(t, aliceClient, paymentChannel, 5)
+	resp = performGetRequest(t, "bytes=0-4", fmt.Sprintf("http://%s/file?channelId=%s&amount=%d&signature=%s", proxyAddress, voucher.ChannelId, voucher.Amount.Int64(), voucher.Signature.ToHexString()))
+	// We want to make sure that the response body is only the first 4 bytes
+	body, statusCode := getResponseInfo(t, resp)
+	if body != testFileContent[0:5] {
+		t.Fatalf("Expected response body to be %s, but got %s", testFileContent[0:5], body)
+	}
+	if statusCode != http.StatusPartialContent {
+		t.Fatalf("Expected status code %d, but got %d", http.StatusPartialContent, statusCode)
+	}
+
+	// It should handle a complex range request
+	// Multipart ranges include extra information about the ranges in the response body
+	const multiPartResponseSize = 346
```

Comment:
The response will look something like this
```
--01a50f63ca7bbec85c50a0b8f753646dd4f33bf5cdae5e3d8d81b472bec5
Content-Range: bytes 0-1/57
Content-Type: text/plain; charset=utf-8

Th
--01a50f63ca7bbec85c50a0b8f753646dd4f33bf5cdae5e3d8d81b472bec5
Content-Range: bytes 3-4/57
Content-Type: text/plain; charset=utf-8

s
--01a50f63ca7bbec85c50a0b8f753646dd4f33bf5cdae5e3d8d81b472bec5--
```

Reply:
Is it true that currently the requester pays for every byte of the response body? If so, does it make sense to consider an approach where the requester only pays for content (and not metadata)?

Reply:
> Is it true that currently the requester pays for every byte of the response body? 

That's true! One thing to note is that only multipart range responses include metadata. If you request data with a single range (IE: `bytes=0-49`) the response body will only contain the 50 bytes you requested and no extraneous metadata. So single part ranges the size of the response body is the size of the requested data.

It's only when a multipart range is requested (IE: `bytes=0-10,40-49`) that the extra metadata is included, and thus you'd end up paying more.  If we wanted to address this I think the payment proxy would have to have parse the range information from the request (like we did in this [PR](https://github.com/statechannels/go-nitro/pull/1493)) so we'd know the size of the data being requested.

However I'm wondering if it's worth the effort since paying slightly more for multipart range requests doesn't seem like the end of the world.  In this test the extra metadata dominates the size but that's because we're requesting 4 bytes of data. In the real world the chunks of data you're requesting will probably be much larger than that so the extra cost won't be as painful. It also seems unlikely that people will use a multipart range request for fetching file data, either you request the file in one go or you single part ranges to request chunks of the file over different requests.

Reply:
All good points. Sounds good and let's merge the code as is.